tensorflow/core/common_runtime/process_function_library_runtime.cc

```diff
@@ -637,6 +636,19 @@ Status ProcessFunctionLibraryRuntime::InstantiateMultiDevice(
   TF_RETURN_IF_ERROR(
       PartitionFunctionGraph(device_set, std::move(graph), &subgraphs));
 
+  for (const auto& pair : subgraphs) {
+    DumpGraph("Before running POST_PARTITIONING passes", pair.second.get());
```

Comment:
Change the line to something like:

   DumpGraph(absl::StrCat("Before running POST_PARTITIONING passes on partition for ", pair.first), 
                        pair.second.get());

to log which device the partition is for.